src/java/org/apache/cassandra/index/sai/disk/v1/Segment.java

```diff
@@ -70,7 +70,7 @@ public Segment(IndexContext indexContext, SSTableContext sstableContext, PerInde
         this.indexFiles = indexFiles;
         this.metadata = metadata;
 
-        this.index = IndexSearcher.open(primaryKeyMapFactory, indexFiles, metadata, sstableContext.indexDescriptor, indexContext);
+        this.index = IndexSearcher.open(primaryKeyMapFactory, indexFiles, metadata, sstableContext.sstable().getId(), sstableContext.indexDescriptor, indexContext);
```

Comment:
The `PrimaryKeyMap.Factory` is tied to a specific `SSTable` so it would make sense to add a `getSSTableId` method to that so we don't have to propagate the `SSTableId` down the stack.